src/components/Integrations/Shopify/Shopify.js

```diff
@@ -1,16 +1,33 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Helmet } from 'react-helmet';
+import { HardcodedShopifyClient } from '../../../services/shopify-client.doubles';
 
-const Shopify = () => (
-  <>
-    <Helmet>
-      <title>Doppler | Shopify</title>
-    </Helmet>
-    <section style={{ width: '100%', padding: '60px 30px' }}>
-      Panel de control | Integraciones y preferencias avanzadas
-      <h1>Panel de Control - Shopify</h1>
-    </section>
-  </>
-);
+const shopifyClient = new HardcodedShopifyClient();
+
+const Shopify = () => {
+  const [shopName, setShopName] = useState('');
+  useEffect(() => {
+    const getData = async () => {
+      const result = await shopifyClient.getShopifyData();
+      if (result.length) {
+        setShopName(result[0].shopName);
+      }
+    };
+    getData();
+  }, []);
```

Comment:
Why do you need `getData` function? I think that it will work in the same way:

```js
  useEffect(() => {
      const result = await shopifyClient.getShopifyData();
      if (result.length) {
        setShopName(result[0].shopName);
      }
  }, []);
```

Reply:
I need to tell the function is async somewhere 
![image](https://user-images.githubusercontent.com/2439363/59364331-15b84580-8d0d-11e9-9669-62d32527b838.png)

Reply:
I've tried adding async into useEffect hook but it also gives error, in this post https://www.robinwieruch.de/react-hooks-fetch-data/#react-hooks-programmatically says that the ways is with a function.

Reply:
Right, I would like to understand why the `useEffect(async () => {` does not work, do you have a screenshot of the error? and if it does not work, could you explain to me in a short way? (I have still pending reading the article)

Reply:
According to the documentation every function annotated with async returns an implicit promise: “The async function declaration defines an asynchronous function, which returns an AsyncFunction object. An asynchronous function is a function which operates asynchronously via the event loop, using an implicit Promise to return its result. “. However, an effect hook should return *nothing or a clean up function*. That’s why you may see the following warning in your developer console log: 07:41:22.910 index.js:1452 Warning: useEffect function must return a cleanup function or nothing. Promises and useEffect(async () => …) are not supported, but you can call an async function inside an effect.. That’s why using async directly in the useEffect function isn’t allowed.

Reply:
Oh, I understand now, thank you very much for the clear explanation!